tests/Fixer/ControlStructure/NoUnneededNestedControlBlockFixerTest.php

```diff
@@ -0,0 +1,577 @@
+function foo(){
+if (%s) {
+
+
+
+}
+}
+',
+                    $combined
+                ),
+                sprintf(
+                    '<?php
+function foo(){
+if ($x) {
+    if (%s) {
+
+    }
+}
+}
+',
+                    $second
+                ),
+            ];
+        }
+
+        yield 'already wrapped precedence' => [
+            '<?php
+if ($foo && $a && ($a xor $b)) {
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($a && ($a xor $b)) {
+
+    }
+}
+',
+        ];
+
+        // multiple flat
+
+        $expected = str_repeat("if (\$foo && \$bar) {\n\n# 1\n\n}\n", 10);
+        $input = str_repeat("if (\$foo) {\n    if (\$bar) {\n# 1\n    }\n}\n", 10);
+
+        yield 'multiple flat "if" case' => [
+            "<?php\n".$expected,
+            "<?php\n".$input,
+        ];
+
+        // multiple nested
+
+        yield 'multiple "if" case' => [
+            '<?php
+if ($foo && $bar1 && $bar2 && $bar3) {
+
+
+
+
+
+
+
+}
+',
+            '<?php
+if ($foo) {
+    if ($bar1) {
+        if ($bar2) {
+            if ($bar3) {
+
+            }
+        }
+    }
+}
+',
+        ];
+
+        yield 'multiple "else" case' => [
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar) {
+
+    } else {
+        if ($bar1) {
+
+        } elseif ($bar2) {
+
+
+        }
+    }
+}
+',
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar) {
+
+    } else {
+        if ($bar1) {
+
+        } else {
+            if ($bar2) {
+            }
+        }
+    }
+}
+',
+        ];
+
+        yield 'multiple "elseif" case' => [
+            '<?php
+if ($foo) {
+
+} elseif($c && $bar) {
+
+        if ($foo1) {
+
+        } elseif($c1 && $bar1) {
+
+                if ($foo2) {
+
+                } elseif($c2 && $bar2) {
+
+
+
+                }
+
+        }
+
+}
+',
+            '<?php
+if ($foo) {
+
+} elseif($c) {
+    if ($bar) {
+        if ($foo1) {
+
+        } elseif($c1) {
+            if ($bar1) {
+                if ($foo2) {
+
+                } elseif($c2) {
+                    if ($bar2) {
+
+                    }
+                }
+            }
+        }
+    }
+}
+',
+        ];
+
+        // do not fix cases
+
+        yield 'alternative syntax' => [
+            '<?php
+    if ($a) echo 1;
+
+    if (true) :
+        $foo = 0;
+    endif;
+
+    if (true):
+        echo 1;
+    else:
+        echo 2;
+    endif;
+
+    if ($bar):
+        echo 1;
+    elseif ($foo):
+        echo 2;
+    endif;
+',
+        ];
+
+        yield [
+            '<?php
+if ($foo) {
+
+} else {
+echo 1;
+    if ($bar) {
+
+    }
+}
+',
+        ];
+
+        yield [
+            '<?php
+if ($foo) {
+
+} else {
+    if ($bar) {
+
+    }
+    echo 1;
+}
+',
+        ];
+
+        yield [
+            '<?php
+
+if ($foo) {
+    if ($bar) {
+
+    }
+} else {
+    echo 1;
+}',
+        ];
+
+        yield [
+            '<?php
+
+if ($foo) {
+    if ($bar) {
+
+    }
+} elseif ($a) {
+    echo 1;
+}',
+        ];
```

Comment:
Please add the following test case (maybe using other assignation operators as well):
```php
        yield 'variable assignations in parent blocks' => [
            '<?php
if ((null !== $foo = foo()) && (null !== $bar = bar()) && null !== $baz = baz()) {
            echo 1;
}',
            '<?php
if (null !== $foo = foo()) {
    if (null !== $bar = bar()) {
        if (null !== $baz = baz()) {
            echo 1;
        }
    }
}',
        ];
```